### DISCUSSION TITLE
Naming Conventions?

### DISCUSSION THREAD

**Discussion by Zach-Attach** [buildingamind/NewbornEmbodiedTuringTest#13] [General]
It seems like our naming conventions are all over the place. We install `nett-benchmarks`. We import `netts`. And our repo is `NewbornEmbodiedTuringTest` which abbreviates to `NETT` not `NETTS` and not even `NETTs` because it's only one test.

I noticed that `nett` [is available to grab on PyPI.](https://pypi.org/search/?q=nett) Personally, I think it would make the most sense to just switch over to `nett` and import `nett`. That would simplify things, but if it's too difficult to change what name we are registered under, maybe we import `nett_benchmarks` or `nettBenchmarks`?

Seems like we have the following options:

1. PyPI: `nett-benchmarks`, Import*: `nett`, Repo: `NewbornEmbodiedTuringTest`

- the least work
- might be confusing to some users
- requires:
  -  merging [this branch](https://github.com/buildingamind/NewbornEmbodiedTuringTest/tree/name-changes)

2. PyPI*: `netts-benchmarks`, Import*: `netts_benchmarks`, Repo*: `NewbornEmbodiedTuringTests` or `NewbornEmbodiedTuringTests-Benchmarks`
- the most work
- has the added benefit of `netts.nett` becoming `netts_benchmarks.nett`
- makes a new set of URLs (old ones will redirect to new URL but might be a little messy)
- requires:
  - making a new PyPI package _**or**_ renaming existing PyPI package
  - a code change
  - [a rename to the repo](https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository) 

3. PyPI*: `netts-benchmarks`, Import: `netts`, Repo*: `NewbornEmbodiedTuringTests` or `NewbornEmbodiedTuringTests-Benchmarks`
- almost as much work as option 2
- might be confusing to some users
- makes a new set of URLs (old ones will redirect to new URL but might be a little messy)
- requires:
  - making a new PyPI package _**or**_ renaming existing PyPI package
  - [a rename to the repo](https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository) (which means a new set of URLs, old ones will redirect to new URL but might be a little messy)

4. PyPI*: `nett`, Import*: `nett`, Repo: `NewbornEmbodiedTuringTest`
- `nett.NETT` would exist (a name change of NETT could fix this but it is also fine as is)
- might have some accidental installs from typos of `pip install netts` which is a different package and vice-versa
- requires:
  - making a new PyPI package _**or**_ renaming existing PyPI package
  - merging [this branch](https://github.com/buildingamind/NewbornEmbodiedTuringTest/tree/name-changes) after changing references to `nett-benchmarks`

**Comment by Zach-Attach** [comment]
@desaibhargav @smwwood @justinnwood @mchivuku I'd love to hear your thoughts.

**Comment by Zach-Attach** [comment]
I've created a branch for this here: https://github.com/buildingamind/NewbornEmbodiedTuringTest/tree/name-changes
So far, I've just changed the following:
`netts` -> `nett`
`.run(dir=` -> `.run(output_dir=`

Also, I was thinking of doing the following:
`nett.NETT` -> `nett.Test`
buuut I'm not sure if that is confusing since it is not a "test" in the traditional coding sense, so I just kept it as is

**Comment by Zach-Attach** [comment]
Completed in #20

**Comment by desaibhargav** [comment]
As per discussion, we are going with `nett` as the package import name (changed from `netts`), while the repository name and the PyPI distribution stays as is.
